pkg/utils/validators.go

```
@@ -0,0 +1,27 @@
+package utils
+
+import (
+	m "book_service/pkg/models/common"
+	"github.com/gin-gonic/gin"
+
+	"github.com/google/uuid"
+)
+
+func IsValidUUID(u string) bool {
+	_, err := uuid.Parse(u)
+	return err == nil
+}
+
+func IsValidRange(price m.PriceRange) bool {
+	return price.Min > price.Max
+}
+
+func GetValidatedPayload[T any](c *gin.Context) T {
+	val, exists := c.Get("validated")
+	if !exists {
+		var zero T
+		return zero
```

If we don't find the query is validated - don't we want to return an error?

---

This function is used **after** the validation logic middleware so this function is just for extraction.

---

So:
1. sounds like this use case cannot really happen, right?
2. If it does (due to a bug?) - do we want to continue with the zero value? Sounds like we will get undesired behavior

---

[DISCUSS]

---

CHECK!

---

I really didn't like my solution because it adds a lot of code:
```go
func GetValidatedPayload[T any](c *gin.Context) (T, error) {
	val, exists := c.Get(consts.ValidatedAccess)
	if !exists {
		var zero T
		return zero, fmt.Errorf("payload not found in context")
	}
	typedVal, _ := val.(T)
	return typedVal, nil
}
```


```go
	bookReq, err := utils.GetValidatedPayload[req.GetBook](c)
	if err != nil {
		log.Errorf("Error getting book by ID: %v", err)
		c.JSON(http.StatusInternalServerError, gin.H{"message": err.Error()})
		return
	}
```